search.py

```diff
@@ -420,8 +420,35 @@ def astar_search(problem, h=None, display=False):
     return best_first_graph_search(problem, lambda n: n.path_cost + h(n), display)
 
 
+def cost_limited_astar_search(problem, limit, f):
```

Comment:
You need to comment the code a bit, unfortunately right now it may be a bit hard to follow.

Reply:
Hey, thanks for the feedback! I added more comments and descriptions for the search function. Please feel free to let me know if they are not clear.